lib/yaml_normalizer/ext/nested.rb

```diff
@@ -17,21 +17,21 @@ module Nested
       #   => {"a"=>{"b"=>{"c"=>1}}, "b"=>{"x"=>2, "y"=>{"ok"=>true}, "z"=>4}}
       # @return [Hash] tree-shaped Hash
       def nested
-        tree = Hash.new { |h, k| h[k] = Hash.new(&h.default_proc) }
+        tree = {}
         each { |key, val| nest_key(tree, key.to_s, val) }
-        tree.default_proc = nil
         tree
       end
 
       private
 
       def nest_key(hash, key, val)
-        if key.include?('.')
-          keys = key.split('.')
-          hash.dig(*keys[0..-2])[keys.fetch(-1)] = val
-        else
-          hash[key] = val
+        keys = key.split('.')
```

Comment:
key.split('.') returns [] if key is "", assigning val to sub_hash[nil]. Not sure if that's a common use case, but I'd prefer assigning val to sub_hash[""]

Reply:
Ok, so I made a `.to_s` at the last_key to get this behaviour and added it to the test case.